src/loaders/OMETiffLoader.js

```diff
@@ -0,0 +1,322 @@
+      }
+      case 'XYTCZ': {
+        return z * SizeT * SizeC + channelIndex * SizeT + time;
+      }
+      case 'XYTZC': {
+        return channelIndex * SizeT * SizeZ + z * SizeT + time;
+      }
+      default: {
+        throw new Error('Dimension order is required for OMETIFF');
+      }
+    }
+  }
+
+  /**
+   * Handles `onTileError` within deck.gl
+   * @param {Error} err Error thrown in tile layer
+   */
+  // eslint-disable-next-line class-methods-use-this
+  onTileError(err) {
+    console.error(err);
+  }
+
+  /**
+   * Returns image tiles at tile-position (x, y) at pyramidal level z.
+   * @param {number} x positive integer
+   * @param {number} y positive integer
+   * @param {number} z positive integer (0 === highest zoom level)
+   * @param {Array} loaderSelection, Array of number Arrays specifying channel selections
+   * @returns {Object} data: TypedArray[], width: number (tileSize), height: number (tileSize)
+   */
+  async getTile({ x, y, z, loaderSelection }) {
+    if (!this._tileInBounds({ x, y, z })) {
+      return null;
+    }
+    const { tiff, offsets, isLegacyBioFormatsPyramid, omexml, tileSize } = this;
+    const { SizeZ, SizeT, SizeC } = omexml;
+    const pyramidOffset = z * SizeZ * SizeT * SizeC;
+    let image;
+    const tileRequests = loaderSelection.map(async index => {
+      const pyramidIndex = pyramidOffset + index;
+      if (isLegacyBioFormatsPyramid) {
+        if (offsets) {
+          tiff.ifdRequests[pyramidIndex] = tiff.parseFileDirectoryAt(
+            offsets[pyramidIndex]
+          );
+        }
+      } else {
+        const parentImage = await tiff.getImage(index);
+        if (z !== 0) {
+          tiff.ifdRequests[pyramidIndex] = tiff.parseFileDirectoryAt(
+            parentImage.fileDirectory.SubIFDs[z - 1]
+          );
+        }
+      }
+      image = await tiff.getImage(pyramidIndex);
+      return this._getChannel({ image, x, y });
+    });
+    const tiles = await Promise.all(tileRequests);
+    return {
+      data: tiles,
+      width: tileSize,
+      height: tileSize
+    };
+  }
+
+  /**
+   * Returns full image panes (at level z if pyramid)
+   * @param {number} z positive integer (0 === highest zoom level)
+   * @param {Array} loaderSelection, Array of number Arrays specifying channel selections
+   * @returns {Object} data: TypedArray[], width: number, height: number
+   */
+  async getRaster({ z, loaderSelection }) {
+    const {
+      tiff,
+      offsets,
+      omexml,
+      isLegacyBioFormatsPyramid,
+      poolOrDecoder
+    } = this;
+    const { SizeZ, SizeT, SizeC } = omexml;
+    const rasters = await Promise.all(
+      loaderSelection.map(async index => {
+        const pyramidIndex = z * SizeZ * SizeT * SizeC + index;
+        if (isLegacyBioFormatsPyramid) {
+          if (offsets) {
+            tiff.ifdRequests[pyramidIndex] = tiff.parseFileDirectoryAt(
+              offsets[pyramidIndex]
+            );
```

Comment:
can you explain what is being mutated here on `tiff`? what is `ifdRequests`?

Reply:
Sure, I'll leave a note.  This allows us to parse images based on their offset within the file.